### PR TITLE
Develop fix

### PR DESCRIPTION
**Pull request by vlodkopits**
# REVIEWERS

- [ ] @sonishc 
- [ ] @StepanKovalyshyn 

# DESCRIPTION

** Travis and Rubocop configuration. Ad new gem to Gemfile and delete duplicate lines. Also make change to User create migration, because db:migrate abort with error, and no db created and Travis don't pass. This is my mistake with user migration, in future make change by other migration, but now it fall **

# TEST COVERAGE
**test for User model**

# EXPERT REVIEW

- [ ] @rokor
- [ ] @okohuch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,10 +5,12 @@ capybara-*.html
 /tmp
 /db/*.sqlite3
 /db/*.sqlite3-journal
+/db/schema.rb
 /public/system
 /coverage/
 /spec/tmp
 *.orig
+Gemfile.lock
```
Comment (omaik) [inline .gitignore]: I am not sure this is good idea, as well as `schema.rb`

File: .rubocop.yml [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
+  Max: 20
+
+Documentation:  
+ Enabled: false
+
+RSpec/MultipleExpectations:
+  Enabled: false
+
+Metrics/BlockLength:
+  Enabled: false
```
Comment (omaik) [inline .rubocop.yml]: New line at the EOF

File: db/migrate/20170921115315_create_users.rb [hunk 3]
```diff
@@ -1,16 +1,13 @@
 class CreateUsers < ActiveRecord::Migration[5.1]
   def change
     create_table :users do |t|
-      t.belongs_to :level, foreign_key: true
-      t.belongs_to :role, foreign_key: true
-
+      t.belongs_to :role, index: true
```
Comment (omaik) [inline db/migrate/20170921115315_create_users.rb:4]: It is not a good approach to edit existing migration, just FYI

File: spec/factories/users.rb [hunk 4]
```diff
@@ -1,7 +1,12 @@
 FactoryGirl.define do
+  factory :role do
+    sequence(:title) { |n| "role#{n}" }
+  end
+
   factory :user do
     sequence(:name) { |n| "User#{n}" }
     sequence(:email) { |n| "user#{n}@test.com" }
     sequence(:password) { |n| "123456#{n}" }
+    role
```
Comment (omaik) [inline spec/factories/users.rb]: user returns role?))

Comment (vlodkopits) [reply spec/factories/users.rb]: https://stackoverflow.com/questions/11530616/how-to-create-belongs-to-relationship-in-factory-girl-sequence is it wrong?

File: db/migrate/20170921115315_create_users.rb (hunk 3 above)
Comment (vlodkopits) [reply db/migrate/20170921115315_create_users.rb:4]: Yes, but it is wrong migration, which doesn't work when run db:migration and crush all other migration. it's my mistake from first PR

Comment (okohuch) [reply db/migrate/20170921115315_create_users.rb:4]: How did you run it the first time then?

Comment (vlodkopits) [reply db/migrate/20170921115315_create_users.rb:4]: with commented relationship, and before commit I delete comment :(

Comment (okohuch) [reply db/migrate/20170921115315_create_users.rb:4]: I already said that it is bad practice to write the code in advance! Migration with comments -- I do not know what to say here...

File: spec/factories/users.rb (hunk 4 above)
Comment (okohuch) [reply spec/factories/users.rb]: It's better to put the role as a parameter while invoking the factory in tests since you will have determined roles.

Comment (rokor) [inline spec/factories/users.rb]: Factory role - good. user Factory returning role - bad.

File: db/migrate/20170921115315_create_users.rb (hunk 3 above)
Comment (rokor) [reply db/migrate/20170921115315_create_users.rb:4]: Situation is catch 22 and Yes it is bad but also we need this so any good suggestion on this one?

**Review by rokor — CHANGES_REQUESTED** [review]

**Review by rokor — APPROVED** [review]

**Pull request merged**